app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,400 @@
+            new Function0<Unit>() {
+              @Override public Unit invoke() {
+                ActivityCompat.requestPermissions(LocalFileTransferActivity.this, new String[] { Manifest.permission.ACCESS_COARSE_LOCATION },
+                    PERMISSION_REQUEST_CODE_COARSE_LOCATION);
+                return Unit.INSTANCE;
+              }
+            });
+      } else {
+        ActivityCompat.requestPermissions(this, new String[] { Manifest.permission.ACCESS_COARSE_LOCATION },
+            PERMISSION_REQUEST_CODE_COARSE_LOCATION);
+      }
+      return false;
+
+    } else {
+      return true; // Control reaches here: Either permission granted at install time, or at the time of request
+    }
+  }
+
+  private boolean checkExternalStorageWritePermission() { // To access and store the zims
+    if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        == PackageManager.PERMISSION_DENIED) {
+
+      if (ActivityCompat.shouldShowRequestPermissionRationale(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+        alertDialogShower.show(KiwixDialog.StoragePermissionRationale.INSTANCE,
+            new Function0<Unit>() {
+              @Override public Unit invoke() {
+                ActivityCompat.requestPermissions(LocalFileTransferActivity.this, new String[] { Manifest.permission.WRITE_EXTERNAL_STORAGE },
+                    PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS);
+                return Unit.INSTANCE;
+              }
+            });
+      } else {
+        ActivityCompat.requestPermissions(this, new String[] { Manifest.permission.WRITE_EXTERNAL_STORAGE },
+            PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS);
+      }
+      return false;
+
+    } else {
+      return true; // Control reaches here: Either permission granted at install time, or at the time of request
+    }
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+      @NonNull int[] grantResults) {
+
+    if (grantResults[0] == PackageManager.PERMISSION_DENIED) {
+      switch (requestCode) {
+
+        case PERMISSION_REQUEST_CODE_COARSE_LOCATION: {
+          Log.e(TAG, "Location permission not granted");
+
+          showToast(this, R.string.permission_refused_location, Toast.LENGTH_LONG);
+          finish();
+          break;
+        }
+
+        case PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS: {
+          Log.e(TAG, "Storage write permission not granted");
+
+          showToast(this, R.string.permission_refused_storage, Toast.LENGTH_LONG);
+          finish();
+          break;
+        }
+
+        default: {
+          super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+          break;
+        }
+      }
+    }
+  }
+
+  private boolean isLocationServiceEnabled() {
+    return (isProviderEnabled(LocationManager.GPS_PROVIDER) || isProviderEnabled(LocationManager.NETWORK_PROVIDER));
```

Comment:
brackets are unnecessary but this is slight enough that I don't care, not required to complete this PR

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/e0651dd88c2ee30b8e17d54ff215384a1d877681#diff-e1d59ae6e15dcb502e125fb6e243d15dR319)